x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -562,6 +563,46 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     return await this.baseClient.removeReferencesTo(type, id, options);
   }
 
+  public async openPointInTimeForType(
+    type: string | string[],
+    options: SavedObjectsOpenPointInTimeOptions
+  ) {
+    try {
+      const args = { type, options };
+      await this.ensureAuthorized(type, 'open_point_in_time', options?.namespace, { args });
+    } catch (error) {
+      this.auditLogger.log(
+        savedObjectEvent({
+          action: SavedObjectAction.OPEN_POINT_IN_TIME,
+          error,
+        })
+      );
+      throw error;
+    }
+
+    const pit = await this.baseClient.openPointInTimeForType(type, options);
+
+    this.auditLogger.log(
+      savedObjectEvent({
+        action: SavedObjectAction.OPEN_POINT_IN_TIME,
+      })
+    );
+
+    return pit;
+  }
+
+  public async closePointInTime(id: string) {
+    const response = await this.baseClient.closePointInTime(id);
```

Comment:
I don't yet have an `ensureAuthorized` here because I didn't add an options object with a `namespace` -- this is still on my todo list

Reply:
Added a `namespace` but realized this method doesn't take a type (which `ensureAuthorized` requires)

Reply:
I'm getting a couple security api integration test failures as well, though it isn't yet clear to me if they are directly related to this.

Reply:
Issues have been fixed here, however as `closePointInTime` doesn't accept any `types`, there isn't a way we can call `ensureAuthorized` in the first place. For now it isn't being called (meaning if you happen to have a PIT ID, I guess you could close it if you wanted to -- not sure if that's a security concern or not).

The alternative would be to require users to also provide a list of types in order to authorize them, even though this method doesn't otherwise need the types for anything. If we took this route, consumers would just need to take care to pass the same types that they originally passed when opening the PIT.

cc @legrego

Reply:
I somehow missed this thread in my initial review. You're absolutely right that we need a SO type in order to perform SO authorization. I think omitting the call to `ensureAuthorized` is acceptable here, and I'll try to justify why:

Elasticsearch only requires `read` privileges on an index in order to open/close a PIT against it. The "equivalent" to having `read` privileges in Kibana is the ability to login at the most basic level. Since we don't have any metadata about open PITS (the indices they apply to, for example), we have no way of knowing what a PIT was used for when we attempt to close it. The best we can do is to make sure the user is _authenticated_ to Kibana, which is always done prior to obtaining an instance of a scoped Saved Objects Client.

An alternative would be to check against the `config` type (which all users have access to), but I don't like this for a couple of reasons:
1) `config` only exists in the primary `.kibana` index. The SO client can be scoped to other indices, to this authorization check would be incomplete at best, and incorrect at worst.
2) It confuses the audit log -- we would record the fact that a PIT was being closed for the `config` SO type, which just doesn't make sense.


Luke, would you mind adding a comment to this function explaining why we are intentionally omitting a call to `ensureAuthorized` here? Otherwise this will look like an omission/mistake to a casual reader.

Reply:
Your reasoning here makes sense to me, thanks for outlining your thoughts on this - will add a comment to make this clear.